MineInAbyss/src/main/java/com/derongan/minecraft/mineinabyss/Ascension/Effect/Effects/PotionAscensionEffect.java

```diff
@@ -17,22 +30,54 @@ public PotionAscensionEffect(int offset, int strength, int duration, int iterati
     @Override
     public void applyEffect(Player player) {
         for(PotionEffectType e:effectsToApply){
-            if (player.getPotionEffect(e) != null)
-                return;
-            player.addPotionEffect(new PotionEffect(e, durationRemaining, strength));
+            if (player.getPotionEffect(e) != null) {
+                mergeExtendPotionEffect(player, e, durationRemaining);
+            } else {
+                player.addPotionEffect(new PotionEffect(e, durationRemaining, strength));
+            }
         }
     }
 
-    //TODO make potion effects be able to merge in different ways
-    /** public void mergeAddPotionEffect(Player player, PotionEffectType potionEffect, int newEffectDuration, int newStrength) {
-             if (player.getPotionEffect(potionEffect).getDuration() < newEffectDuration) {
-                 } else if (player.getPotionEffect(potionEffect).getDuration() > newEffectDuration) {
-                 } else {}
-                 }**/
+    //potion effect merge shenanigans, creates a new stronger effect for when the effects overlap and another effect for the longer effect
+    //TODO I have no idea why it doesn't work
```

Comment:
Thats not something I like to see in a merge request :P... Is this comment current?

Reply:
Yes, you do remember how long i tried to make this work.
Probably has some weird code in it when i tried to get it to work, but at least somebody else can.

Reply:
I'm against merging in broken code to master